modules/phase_field/src/userobjects/DiscreteNucleationInserter.C

```diff
@@ -78,13 +81,18 @@ DiscreteNucleationInserter::threadJoin(const UserObject &y)
 {
   // combine _local_nucleus_list entries from all threads on the current process
   const DiscreteNucleationInserter & uo = static_cast<const DiscreteNucleationInserter &>(y);
-  _local_nucleus_list.insert(_local_nucleus_list.end(), uo._local_nucleus_list.begin(), uo._local_nucleus_list.end());
+  _global_nucleus_list.insert(_global_nucleus_list.end(), uo._local_nucleus_list.begin(), uo._local_nucleus_list.end());
+  _changes_made += uo._changes_made;
 }
 
 void
 DiscreteNucleationInserter::finalize()
 {
-  // here we need to combine all _local_nucleus_list into a the _global_nucleus_list
-  _global_nucleus_list = _local_nucleus_list;
+  // add the _local_nucleus_list of thread zero
+  _global_nucleus_list.insert(_global_nucleus_list.end(), _local_nucleus_list.begin(), _local_nucleus_list.end());
+  // combine _global_nucleus_lists from all MPI ranks
   _communicator.allgather(_global_nucleus_list);
```

Comment:
The error is right here @jwpeterson, allgather should turn ```_global_nucleus_list``` into a vector that is a concatenation of all local ```_global_nucleus_list``` vectors. This works on my machine, but on the build box, I get only the entries from processor 0 and the rest filled with zeroes or undefined values. I output the local copies of ```_global_nucleus_list``` before the allgather and they contain the correct values. What could cause allgather to fail?!